lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -407,18 +410,14 @@ def add_payment_method(xml, amount, payment_method, options)
               xml.session 'shopperIPAddress' => options[:ip] if options[:ip]
               xml.session 'id' => options[:session_id] if options[:session_id]
             end
-
-            if three_d_secure = options[:three_d_secure]
-              add_three_d_secure(three_d_secure, xml)
-            end
           end
         end
       end
 
       def add_three_d_secure(three_d_secure, xml)
         xml.info3DSecure do
           xml.threeDSVersion three_d_secure[:version]
-          if /^2/.match?(three_d_secure[:version])
+          if three_d_secure[:version] && three_d_secure[:ds_transaction_id]
```

Comment:
There are no test to modify or add for this code change?

Reply:
Hi Bertrand - This will be covered under the existing test_3ds_version_2_request and test_3ds_version_1_request. Also, if transaction get into the 3D secure, version and transaction Ids are must have values from MPI.

Reply:
Thank you Chandan! I think there are still indentation issues, likely tab vs space. `bundle exec rubocop` fails. Can you reproduce this? If, yes, can you update the code so rubocop does not fail?


```
github.com/activemerchant/active_merchant (4017)$ bundle exec rubocop lib/active_merchant/billing/gateways/worldpay.rb
Inspecting 1 file
W

Offenses:

lib/active_merchant/billing/gateways/worldpay.rb:216:1: C: Layout/Tab: Tab detected. (https://github.com/rubocop-hq/ruby-style-guide#spaces-indentation)
			        if three_d_secure = options[:three_d_secure]
^^^
lib/active_merchant/billing/gateways/worldpay.rb:216:12: C: Layout/IndentationConsistency: Inconsistent indentation detected. (https://github.com/rubocop-hq/ruby-style-guide#spaces-indentation)
			        if three_d_secure = options[:three_d_secure] ...
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/active_merchant/billing/gateways/worldpay.rb:217:12: C: Layout/IndentationWidth: Use 2 (not 5) spaces for indentation. (https://github.com/rubocop-hq/ruby-style-guide#spaces-indentation)
                add_three_d_secure(three_d_secure, xml)
           ^^^^^
lib/active_merchant/billing/gateways/worldpay.rb:218:15: W: Layout/EndAlignment: end at 218, 14 is not aligned with if at 216, 11.
              end
              ^^^
```

Reply:
Hi Bertrand - Yes, I was able to reproduce the issue and have fixed the same now. Please merge the PR at the earliest.